### PR TITLE
Try adopting withContiguousStorageIfAvailable in Array and ContiguousArray append(contentsOf:)

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/Array.swift [hunk 1]
```diff
@@ -1197,6 +1224,16 @@ extension Array: RangeReplaceableCollection {
   @_semantics("array.append_contentsOf")
   public mutating func append<S: Sequence>(contentsOf newElements: __owned S)
     where S.Element == Element {
+      
+    //let wasCollection = newElements.withContiguousStorageIfAvailable {
```
Comment (3405691582) [inline stdlib/public/core/Array.swift]: Cleanup?

Comment (3405691582) [reply stdlib/public/core/Array.swift]: Ah, guessing perhaps you might still be playing with this.

**Comment by eeckstein** [comment]
Can you also check the code size impact on the source compatibility suite?

File: stdlib/public/core/Array.swift [hunk 2]
```diff
@@ -1176,7 +1176,34 @@ extension Array: RangeReplaceableCollection {
     _reserveCapacityAssumingUniqueBuffer(oldCount: oldCount)
     _appendElementAssumeUniqueAndCapacity(oldCount, newElement: newElement)
   }
+  
+  @inlinable
```
Comment (eeckstein) [inline stdlib/public/core/Array.swift]: This needs to be @_alwaysEmitIntoClient to not break ABI.

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**